### ISSUE TITLE
SDK 3 Working project

### ISSUE THREAD

**Issue by grillorafael** [pebble/pebblejs#31]
Hi guys!

I'm trying to make this work on SDK 3 but it is not working.
1. pebble convert-project
  1. Pebble generated a new wscript that I merged with the old one

When I build the project I get lots of errors in simply. It seems that those errors are related to bitmap and colors.

**Comment by idoodler** [comment]
I am having the same issue and I think this is a serious problem! Or did we do anything wrong?

**Comment by grillorafael** [comment]
@idoodler, I think this has something to do with the new SDK version and not with the Pebble.JS itself.

Simply has to adapt to the new SDK version.

**Comment by Meiguro** [comment]
Yes, that is correct, SDK 3 was such a surprise, it was a surprise to me as well! I will update pebble.js as soon as I get the chance. In addition to C changes, there may also be some JavaScript incompatibilities that I will have to work around.

**Comment by grillorafael** [comment]
If you need any help please ask me!

I'm an amateur using pebble but I can try :stuck_out_tongue:

**Comment by Katharine** [comment]
There are presently no changes to PebbleKit JS in SDK 3.

**Comment by Meiguro** [comment]
@Katharine ah, well the surprise for me is gone! It should be simple then.

**Comment by Meiguro** [comment]
For tinkerers, `feature-sdk-3` has the bare minimum source changes to make it work with SDK 3. I'll continue by adding better color support.

@Katharine @sarfata Let me know if it is safe or desirable for CloudPebble to have this branch merged into master, and I will do so!

**Comment by grillorafael** [comment]
Thanks @Meiguro! I'll give it a try

**Comment by grillorafael** [comment]
Build and install working. The demo app is also working @Meiguro

**Comment by idoodler** [comment]
@Meiguro The wscript and the appinfo have been adapted automatically for me. But to make sure, that there is no error I replaced my wscript with the one you provided, I also checked the appinfo.json and there is no significant change, so I did not touched it. I then builded the project and I got a huge load of errors. I think there is another error.

Here is my terminal output:

"Last login: Sun Mar  1 19:26:49 on ttys000
You have mail.
Davids-MBP:~ David$ cd /Users/David/Downloads/pebblejs-master
Davids-MBP:pebblejs-master David$ pebble build
Setting top to                           : /Users/David/Downloads/pebblejs-master 
Setting out to                           : /Users/David/Downloads/pebblejs-master/build 
Found Pebble SDK for aplite in:          : /Users/David/pebble-dev/PebbleSDK-3.0-dp1/Pebble/aplite 
Checking for program gcc,cc              : arm-none-eabi-gcc 
Checking for program ar                  : arm-none-eabi-ar 
Found Pebble SDK for basalt in:          : /Users/David/pebble-dev/PebbleSDK-3.0-dp1/Pebble/basalt 
Checking for program gcc,cc              : arm-none-eabi-gcc 
Checking for program ar                  : arm-none-eabi-ar 
'configure' finished successfully (0.056s)
Waf: Entering directory `/Users/David/Downloads/pebblejs-master/build'
[17/73] app_resources.pbpack.manifest: build/resources/aplite/fonts/UbuntuMono-Regular.ttf.MONO_FONT_14.pfo build/resources/aplite/images/logo_splash.pbi build/resources/aplite/images/LoxoneForPebble.pbi build/resources/aplite/images/minus.pbi build/resources/aplite/images/plus.pbi build/resources/aplite/images/push.pbi build/resources/aplite/images/up.pbi build/resources/aplite/images/info.pbi build/resources/aplite/images/close.pbi build/resources/aplite/images/error.pbi build/resources/aplite/images/success.pbi build/resources/aplite/images/warning.pbi ../../pebble-dev/PebbleSDK-3.0-dp1/Pebble/common/tools/pbpack_meta_data.py -> build/aplite/app_resources.pbpack.manifest
[36/73] app_resources.pbpack.manifest: build/resources/basalt/fonts/UbuntuMono-Regular.ttf.MONO_FONT_14.pfo build/resources/basalt/images/logo_splash.png build/resources/basalt/images/LoxoneForPebble.png build/resources/basalt/images/minus.png build/resources/basalt/images/plus.png build/resources/basalt/images/push.png build/resources/basalt/images/up.png build/resources/basalt/images/info.png build/resources/basalt/images/close.png build/resources/basalt/images/error.png build/resources/basalt/images/success.png build/resources/basalt/images/warning.png ../../pebble-dev/PebbleSDK-3.0-dp1/Pebble/common/tools/pbpack_meta_data.py -> build/basalt/app_resources.pbpack.manifest
[39/73] c: src/simply/simply_accel.c -> build/src/simply/simply_accel.c.39.o
[40/73] c: src/simply/simply_menu.c -> build/src/simply/simply_menu.c.39.o
[43/73] c: src/simply/simply_splash.c -> build/src/simply/simply_splash.c.39.o
[44/73] c: src/simply/simply_stage.c -> build/src/simply/simply_stage.c.39.o
[48/73] c: src/simply/simply_wakeup.c -> build/src/simply/simply_wakeup.c.39.o
[48/73] c: src/simply/simply_window.c -> build/src/simply/simply_window.c.39.o
../src/simply/simply_accel.c: In function 'send_accel_tap':
../src/simply/simply_accel.c:42:5: error: missing initializer [-Werror=missing-field-initializers]
../src/simply/simply_accel.c:42:5: error: (near initialization for 'packet.packet.length') [-Werror=missing-field-initializers]
cc1: all warnings being treated as errors
../src/simply/simply_splash.c: In function 'window_load':
../src/simply/simply_splash.c:28:46: error: 'RESOURCE_ID_IMAGE_TILE_SPLASH' undeclared (first use in this function)
../src/simply/simply_splash.c:28:46: note: each undeclared identifier is reported only once for each function it appears in
../src/simply/simply_menu.c: In function 'send_menu_item':
../src/simply/simply_menu.c:93:5: error: missing initializer [-Werror=missing-field-initializers]
../src/simply/simply_menu.c:93:5: error: (near initialization for 'packet.packet.length') [-Werror=missing-field-initializers]
../src/simply/simply_menu.c: In function 'request_menu_section':
../src/simply/simply_menu.c:198:3: error: missing initializer [-Werror=missing-field-initializers]
../src/simply/simply_menu.c:198:3: error: (near initialization for '(anonymous).<anonymous>.<anonymous>.title') [-Werror=missing-field-initializers]
../src/simply/simply_menu.c: In function 'request_menu_item':
../src/simply/simply_menu.c:211:5: error: missing initializer [-Werror=missing-field-initializers]
../src/simply/simply_menu.c:211:5: error: (near initialization for '(anonymous).<anonymous>.<anonymous>.title') [-Werror=missing-field-initializers]
../src/simply/simply_menu.c: In function 'handle_menu_section_packet':
../src/simply/simply_menu.c:417:5: error: missing initializer [-Werror=missing-field-initializers]
../src/simply/simply_menu.c:417:5: error: (near initialization for '(anonymous).<anonymous>.<anonymous>.title') [-Werror=missing-field-initializers]
../src/simply/simply_menu.c: In function 'handle_menu_item_packet':
../src/simply/simply_menu.c:428:5: error: missing initializer [-Werror=missing-field-initializers]
../src/simply/simply_menu.c:428:5: error: (near initialization for '(anonymous).<anonymous>.<anonymous>.title') [-Werror=missing-field-initializers]
../src/simply/simply_menu.c: In function 'simply_menu_create':
../src/simply/simply_menu.c:480:5: error: missing initializer [-Werror=missing-field-initializers]
../src/simply/simply_menu.c:480:5: error: (near initialization for '(anonymous).window.window') [-Werror=missing-field-initializers]
cc1: all warnings being treated as errors
../src/simply/simply_wakeup.c: In function 'send_launch_reason':
../src/simply/simply_wakeup.c:52:5: error: missing initializer [-Werror=missing-field-initializers]
../src/simply/simply_wakeup.c:52:5: error: (near initialization for 'packet.packet.length') [-Werror=missing-field-initializers]
../src/simply/simply_wakeup.c: In function 'send_wakeup_signal':
../src/simply/simply_wakeup.c:63:5: error: missing initializer [-Werror=missing-field-initializers]
../src/simply/simply_wakeup.c:63:5: error: (near initialization for 'packet.packet.length') [-Werror=missing-field-initializers]
cc1: all warnings being treated as errors
../src/simply/simply_stage.c: In function 'send_animate_element_done':
../src/simply/simply_stage.c:117:5: error: missing initializer [-Werror=missing-field-initializers]
../src/simply/simply_stage.c:117:5: error: (near initialization for 'packet.packet.length') [-Werror=missing-field-initializers]
../src/simply/simply_stage.c: In function 'simply_stage_create':
../src/simply/simply_stage.c:625:3: error: missing initializer [-Werror=missing-field-initializers]
../src/simply/simply_stage.c:625:3: error: (near initialization for '(anonymous).window.window') [-Werror=missing-field-initializers]
cc1: all warnings being treated as errors
../src/simply/simply_window.c: In function 'send_click':
../src/simply/simply_window.c:56:5: error: missing initializer [-Werror=missing-field-initializers]
../src/simply/simply_window.c:56:5: error: (near initialization for 'packet.packet.length') [-Werror=missing-field-initializers]
cc1: all warnings being treated as errors
Waf: Leaving directory `/Users/David/Downloads/pebblejs-master/build'
Build failed
 -> task in 'aplite/pebble-app.elf' failed (exit status 1): 
	{task 4571664720: c simply_accel.c -> simply_accel.c.39.o}
['arm-none-eabi-gcc', '-std=c99', '-mcpu=cortex-m3', '-mthumb', '-ffunction-sections', '-fdata-sections', '-g', '-Os', '-Wall', '-Wextra', '-Werror', '-Wno-unused-parameter', '-Wno-error=unused-function', '-Wno-error=unused-variable', '-fPIE', '-I/Users/David/Downloads/pebble-dev/PebbleSDK-3.0-dp1/Pebble/aplite/include', '-I/Users/David/pebble-dev/PebbleSDK-3.0-dp1/Pebble/aplite/include', '-I/Users/David/Downloads/pebblejs-master/build', '-I/Users/David/Downloads/pebblejs-master', '-I/Users/David/Downloads/pebblejs-master/build/src', '-I/Users/David/Downloads/pebblejs-master/src', '-I/Users/David/Downloads/pebblejs-master/build/aplite', '-I/Users/David/Downloads/pebblejs-master/aplite', '-DRELEASE', '-DPBL_PLATFORM_APLITE', '-DPBL_BW', '-D__FILE_NAME__="simply_accel.c"', '../src/simply/simply_accel.c', '-c', '-o', 'src/simply/simply_accel.c.39.o']
 -> task in 'aplite/pebble-app.elf' failed (exit status 1): 
	{task 4571665296: c simply_splash.c -> simply_splash.c.39.o}
['arm-none-eabi-gcc', '-std=c99', '-mcpu=cortex-m3', '-mthumb', '-ffunction-sections', '-fdata-sections', '-g', '-Os', '-Wall', '-Wextra', '-Werror', '-Wno-unused-parameter', '-Wno-error=unused-function', '-Wno-error=unused-variable', '-fPIE', '-I/Users/David/Downloads/pebble-dev/PebbleSDK-3.0-dp1/Pebble/aplite/include', '-I/Users/David/pebble-dev/PebbleSDK-3.0-dp1/Pebble/aplite/include', '-I/Users/David/Downloads/pebblejs-master/build', '-I/Users/David/Downloads/pebblejs-master', '-I/Users/David/Downloads/pebblejs-master/build/src', '-I/Users/David/Downloads/pebblejs-master/src', '-I/Users/David/Downloads/pebblejs-master/build/aplite', '-I/Users/David/Downloads/pebblejs-master/aplite', '-DRELEASE', '-DPBL_PLATFORM_APLITE', '-DPBL_BW', '-D__FILE_NAME__="simply_splash.c"', '../src/simply/simply_splash.c', '-c', '-o', 'src/simply/simply_splash.c.39.o']
 -> task in 'aplite/pebble-app.elf' failed (exit status 1): 
	{task 4571664848: c simply_menu.c -> simply_menu.c.39.o}
['arm-none-eabi-gcc', '-std=c99', '-mcpu=cortex-m3', '-mthumb', '-ffunction-sections', '-fdata-sections', '-g', '-Os', '-Wall', '-Wextra', '-Werror', '-Wno-unused-parameter', '-Wno-error=unused-function', '-Wno-error=unused-variable', '-fPIE', '-I/Users/David/Downloads/pebble-dev/PebbleSDK-3.0-dp1/Pebble/aplite/include', '-I/Users/David/pebble-dev/PebbleSDK-3.0-dp1/Pebble/aplite/include', '-I/Users/David/Downloads/pebblejs-master/build', '-I/Users/David/Downloads/pebblejs-master', '-I/Users/David/Downloads/pebblejs-master/build/src', '-I/Users/David/Downloads/pebblejs-master/src', '-I/Users/David/Downloads/pebblejs-master/build/aplite', '-I/Users/David/Downloads/pebblejs-master/aplite', '-DRELEASE', '-DPBL_PLATFORM_APLITE', '-DPBL_BW', '-D__FILE_NAME__="simply_menu.c"', '../src/simply/simply_menu.c', '-c', '-o', 'src/simply/simply_menu.c.39.o']
 -> task in 'aplite/pebble-app.elf' failed (exit status 1): 
	{task 4571665680: c simply_wakeup.c -> simply_wakeup.c.39.o}
['arm-none-eabi-gcc', '-std=c99', '-mcpu=cortex-m3', '-mthumb', '-ffunction-sections', '-fdata-sections', '-g', '-Os', '-Wall', '-Wextra', '-Werror', '-Wno-unused-parameter', '-Wno-error=unused-function', '-Wno-error=unused-variable', '-fPIE', '-I/Users/David/Downloads/pebble-dev/PebbleSDK-3.0-dp1/Pebble/aplite/include', '-I/Users/David/pebble-dev/PebbleSDK-3.0-dp1/Pebble/aplite/include', '-I/Users/David/Downloads/pebblejs-master/build', '-I/Users/David/Downloads/pebblejs-master', '-I/Users/David/Downloads/pebblejs-master/build/src', '-I/Users/David/Downloads/pebblejs-master/src', '-I/Users/David/Downloads/pebblejs-master/build/aplite', '-I/Users/David/Downloads/pebblejs-master/aplite', '-DRELEASE', '-DPBL_PLATFORM_APLITE', '-DPBL_BW', '-D__FILE_NAME__="simply_wakeup.c"', '../src/simply/simply_wakeup.c', '-c', '-o', 'src/simply/simply_wakeup.c.39.o']
 -> task in 'aplite/pebble-app.elf' failed (exit status 1): 
	{task 4571665360: c simply_stage.c -> simply_stage.c.39.o}
['arm-none-eabi-gcc', '-std=c99', '-mcpu=cortex-m3', '-mthumb', '-ffunction-sections', '-fdata-sections', '-g', '-Os', '-Wall', '-Wextra', '-Werror', '-Wno-unused-parameter', '-Wno-error=unused-function', '-Wno-error=unused-variable', '-fPIE', '-I/Users/David/Downloads/pebble-dev/PebbleSDK-3.0-dp1/Pebble/aplite/include', '-I/Users/David/pebble-dev/PebbleSDK-3.0-dp1/Pebble/aplite/include', '-I/Users/David/Downloads/pebblejs-master/build', '-I/Users/David/Downloads/pebblejs-master', '-I/Users/David/Downloads/pebblejs-master/build/src', '-I/Users/David/Downloads/pebblejs-master/src', '-I/Users/David/Downloads/pebblejs-master/build/aplite', '-I/Users/David/Downloads/pebblejs-master/aplite', '-DRELEASE', '-DPBL_PLATFORM_APLITE', '-DPBL_BW', '-D__FILE_NAME__="simply_stage.c"', '../src/simply/simply_stage.c', '-c', '-o', 'src/simply/simply_stage.c.39.o']
 -> task in 'aplite/pebble-app.elf' failed (exit status 1): 
	{task 4571665808: c simply_window.c -> simply_window.c.39.o}
['arm-none-eabi-gcc', '-std=c99', '-mcpu=cortex-m3', '-mthumb', '-ffunction-sections', '-fdata-sections', '-g', '-Os', '-Wall', '-Wextra', '-Werror', '-Wno-unused-parameter', '-Wno-error=unused-function', '-Wno-error=unused-variable', '-fPIE', '-I/Users/David/Downloads/pebble-dev/PebbleSDK-3.0-dp1/Pebble/aplite/include', '-I/Users/David/pebble-dev/PebbleSDK-3.0-dp1/Pebble/aplite/include', '-I/Users/David/Downloads/pebblejs-master/build', '-I/Users/David/Downloads/pebblejs-master', '-I/Users/David/Downloads/pebblejs-master/build/src', '-I/Users/David/Downloads/pebblejs-master/src', '-I/Users/David/Downloads/pebblejs-master/build/aplite', '-I/Users/David/Downloads/pebblejs-master/aplite', '-DRELEASE', '-DPBL_PLATFORM_APLITE', '-DPBL_BW', '-D__FILE_NAME__="simply_window.c"', '../src/simply/simply_window.c', '-c', '-o', 'src/simply/simply_window.c.39.o']
[ERROR   ] A compilation error occurred
Davids-MBP:pebblejs-master David$ 
"

**Comment by Meiguro** [comment]
@idoodler, it looks like you might be accidentally using the default wscript. For example, the -Wno-error=missing-field-initializers cflag in pebble.js' wscript is missing your compilation. I was able to reproduce your problem by creating a new pebble project and overwriting pebble.js' wscript with the newly generated one. pebble.js uses a custom wscript, so it's very important to use a wscript that is a derivative of pebble.js' rather than the SDK template. I've not yet merged the new wscript to `master`, did you checkout from the branch `feature-sdk-3`? I recommend directly copying the wscript and overwriting instead of resolving merge conflicts in this case unless you made some modifications.

**Comment by idoodler** [comment]
@Meiguro Thanks for helping me, everything is working now:)

**Comment by Meiguro** [comment]
You're welcome @idoodler!

I've further updated `feature-sdk-3` to be compatible with the new SDK 3.0-dp2 and the old SDK 2.9. SDK 3.0-dp2 introduces helpful macros which conflict with static inline functions in the branch that serve the same purpose, so a bit of resolution was necessary.

**Comment by mihirt** [comment]
Is pebblejs working with dp3? Tried build just now on most recent version of this repo, got same errors as  @idoodler 
Should i revert to dp2?

**Comment by youtux** [comment]
Here no problems with dp4.

**Comment by idoodler** [comment]
@mihirt If you are using WebStorm or any other IDE with a build in console you may need to execute the PebbleTools via the Terminal.app on your Mac. (Same on Linux??)

You can try it out, it may help you...

**Comment by valexandersaulys** [comment]
Hey I still get this issue! I'm using the most recent pull from here and am trying to load up an image. When I copy-pasted the app.js file into Cloudpebble, it worked fine. However, here on my linux machine its not.  The snippet of error that is returned I pasted below
```
[29/78] c: src/simply/simply_splash.c -> build/src/simply/simply_splash.c.17.o
[56/78] c: src/simply/simply_splash.c -> build/src/simply/simply_splash.c.18.o
[64/78] c: src/simply/simply_splash.c -> build/src/simply/simply_splash.c.19.o
[64/78] c: build/aplite/appinfo.auto.c -> build/aplite/appinfo.auto.c.19.o
[65/78] Start bundling: 
../src/simply/simply_splash.c: In function 'window_load':
../src/simply/simply_splash.c:28:46: error: 'RESOURCE_ID_IMAGE_TILE_SPLASH' undeclared (first use i
n this function)                                                                                  
../src/simply/simply_splash.c:28:46: note: each undeclared identifier is reported only once for eac
h function it appears in                                                                          
../src/simply/simply_splash.c: In function 'window_load':
../src/simply/simply_splash.c:28:46: error: 'RESOURCE_ID_IMAGE_TILE_SPLASH' undeclared (first use i
n this function)                                                                                  
../src/simply/simply_splash.c:28:46: note: each undeclared identifier is reported only once for eac
h function it appears in                                                                          
../src/simply/simply_splash.c: In function 'window_load':
../src/simply/simply_splash.c:28:46: error: 'RESOURCE_ID_IMAGE_TILE_SPLASH' undeclared (first use i
n this function)                                                                                  
../src/simply/simply_splash.c:28:46: note: each undeclared identifier is reported only once for eac
h function it appears in    
```

**Comment by valexandersaulys** [comment]
might this stem from having to having a menu icon?

**Comment by valexandersaulys** [comment]
apparently if I make everything in appinfo.json refer to the same image, I can cut out the default png images. However, I still need to keep the default entries. Weird.

**Comment by valexandersaulys** [comment]
I still have issues though. It seems that prior data is referring to some other entry or older version of the view. app.js never gets read, so no matter how I change it I get an ugly repeating tile view of the splash logo.
